src/shared/settings.ts

```diff
@@ -594,17 +594,18 @@ export class DevSettings extends Settings.define('aws.dev', DEV_SETTINGS) {
  * Simple utility function to 'migrate' a setting from one key to another.
  *
  * Currently only used for simple migrations where we are not concerned about maintaining the
- * legacy definition.
+ * legacy definition. Only migrates to global settings.
```

Comment:
We don't talk about it very often but users could potentially define workspace-specific settings. We shouldn't be automatically editing those files though so it's kind of tricky to migrate.

Another thing to note is that by reading from settings with `get`, we read from the merged view. This is okay so long as you don't try to update the setting. Updating the setting means you're assuming that it's located exclusively in a single scope, but it might not be...

We should restrict all settings that make this assumption to use the `application` or `machine` scope.